Python/compile.c

```diff
@@ -6468,16 +6523,35 @@ compiler_annassign(struct compiler *c, stmt_ty s)
         if (s->v.AnnAssign.simple &&
             (c->u->u_scope_type == COMPILER_SCOPE_MODULE ||
              c->u->u_scope_type == COMPILER_SCOPE_CLASS)) {
-            if (c->c_future.ff_features & CO_FUTURE_ANNOTATIONS) {
-                VISIT(c, annexpr, s->v.AnnAssign.annotation)
+            if (future_annotations) {
+                if (future_annotations) {
+                    VISIT(c, annexpr, s->v.AnnAssign.annotation);
+                }
+                else {
+                    VISIT(c, expr, s->v.AnnAssign.annotation);
+                }
```

Comment:
isn't this dead code? we are already inside `if (future_annotations)` here

Reply:
Right, this is a leftover from trying to implement PEP 649's REPL semantics.